scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,51 +97,119 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
 
-        List<ScannerContext> scannerContextList = new ArrayList<>();
+        PackageResolution packageResolution = project.currentPackage().getResolution();
+        ResolvedPackageDependency rootPkgNode = new ResolvedPackageDependency(project.currentPackage(),
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            PackageOrg org = pkgManifest.org();
+            PackageName name = pkgManifest.name();
+            if (pkgManifest.compilerPluginDescriptor().isEmpty() || !analyzerDescriptors
+                    .contains(org.value() + FORWARD_SLASH + name.value())) {
+                continue;
+            }
 
-        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<String> jarPaths = new ArrayList<>();
+            pluginDesc.dependencies().forEach(dependency -> {
+                jarPaths.add(dependency.getPath());
+            });
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            List<URL> jarUrls = new ArrayList<>();
+            jarPaths.forEach(jarPath -> {
+                try {
+                    URL jarUrl = Path.of(jarPath).toUri().toURL();
+                    jarUrls.add(jarUrl);
+                } catch (MalformedURLException ex) {
+                    throw new RuntimeException(ex);
+                }
+            });
+
+            URLClassLoader ucl = AccessController.doPrivileged((PrivilegedAction<URLClassLoader>) () ->
+                    new URLClassLoader(jarUrls.toArray(new URL[0]), this.getClass().getClassLoader())
+            );
+            InputStream resourceAsStream = ucl.getResourceAsStream(RULES_FILE);
+            if (resourceAsStream == null) {
+                continue;
+            }
+
+            StringBuilder stringBuilder = new StringBuilder();
+            try (BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(resourceAsStream,
+                            StandardCharsets.UTF_8))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    stringBuilder.append(line);
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
You can make this more concise by doing a [mutable reduction](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collector.html), instead of explicitly using a `StringBuilder` and converting to String. Do note this is using the stream API, other than that this should perform more or less the same
```suggestion
            String res;
            try (BufferedReader reader = new BufferedReader(
                    new InputStreamReader(resourceAsStream,
                            StandardCharsets.UTF_8))) {
                res = reader.lines().collect(Collectors.joining());
            } catch (IOException ex) {
                throw new UncheckedIOException(ex);
            }
```